lib/_tls_wrap.js

```diff
@@ -817,51 +843,24 @@ exports.connect = function(/* [port, host], options, cb */) {
       context = tls.createSecureContext(options);
   tls.convertNPNProtocols(options.NPNProtocols, NPN);
 
-  // Wrapping TLS socket inside another TLS socket was requested -
-  // create legacy secure pair
   var socket;
-  var legacy;
   var result;
-  if (options.socket instanceof TLSSocket) {
-    debug('legacy connect');
-    legacy = true;
-    socket = legacyConnect(hostname, options, NPN, context);
-    result = socket.cleartext;
-  } else {
-    legacy = false;
-    socket = new TLSSocket(options.socket, {
-      secureContext: context,
-      isServer: false,
-      requestCert: true,
-      rejectUnauthorized: options.rejectUnauthorized,
-      session: options.session,
-      NPNProtocols: NPN.NPNProtocols,
-      requestOCSP: options.requestOCSP
-    });
-    result = socket;
-  }
-
-  if (socket._handle && !socket._connecting) {
-    onHandle();
-  } else {
-    // Not even started connecting yet (or probably resolving dns address),
-    // catch socket errors and assign handle.
-    if (!legacy && options.socket) {
-      options.socket.once('connect', function() {
-        assert(options.socket._handle);
-        socket._handle = options.socket._handle;
-        socket._handle.owner = socket;
-        socket.emit('connect');
-      });
-    }
-    socket.once('connect', onHandle);
-  }
+  socket = new TLSSocket(options.socket, {
```

Comment:
there a reason this isn't just `var socket =` instead of forward declaring it 2 lines above? Same for `var result` which could be collapsed